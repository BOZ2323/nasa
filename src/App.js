import React, { useEffect, useState } from "react";
import DailySpacePic from "./components/DailySpacePic";

const API_KEY = "kJf4F8gQTewDqd17XKGl4oHuXMu5QC6i0ecPYWCP";

function App() {
  const [data, setData] = useState();

  useEffect(() => {
    getData();
  }, []);

  const getData = async () => {
    try {
      // const response = await fetch(`https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`);
      const response = await fetch(`https://api.nasa.gov/planetary/apod?api_key=${API_KEY}&start_date=2017-07-08&end_date=2017-07-10`);
      const data = await response.json();
      console.log("response", data);
      setData(data);
    } catch {
      console.log("API request failed");
    }
    console.log("data", data);
  };
  return (
    <>
      {/* {data && <DailySpacePic data={data} />} */}
      {data?.map((data) => (
            <DailySpacePic data={data} key={data.title} />
          ))}
    </>
  );
}

export default App;
