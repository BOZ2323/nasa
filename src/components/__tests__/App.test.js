// import * as App from "../../App";
import App, { GetDailyData } from "../../App";
import ReactDOM from 'react-dom'
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

let container;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  
  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });


test("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<App />, div)
})

test("matches snapshot", () => {
    const snap = renderer.create(<App />).toJSON();
    expect(snap).toMatchSnapshot();
})
// describe("Button component", () => {
//     test("it shows the expected text when clicked", () => {
//       act(() => {
//         ReactDOM.render(<GetDailyData text="See last weeks pictures" />, container);
//       });
//       // more soon
//     });
//   });

//not permanent solution - this gets rid of the errormessage that getContext is not implemented
HTMLCanvasElement.prototype.getContext = () => { 

};
