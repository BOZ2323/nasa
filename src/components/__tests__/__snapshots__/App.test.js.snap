// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <main
    className="sc-iqAclL ffNEnI"
  >
    <section
      className="sc-bdnxRM iaMMfV"
      data-testid="test-2"
    >
      <h1
        className="sc-gtsrHT RUswA"
      >
        NASA
      </h1>
      <h1
        className="sc-dlnjwi ctiA-dX"
      >
        Daily Astronomy picture of the day
      </h1>
    </section>
    <button
      className="sc-bqGGPW eqnbAc"
      onClick={[Function]}
    >
      See last weeks pictures
    </button>
    <p
      className="sc-dIsUp iAHDsu"
    >
      Choose here a certain date and you will see that date's picture of the day! 
    </p>
    <section
      className="sc-crzoAE fJQpxT"
    >
      <div
        className="react-date-picker react-date-picker--closed react-date-picker--enabled"
        onFocus={[Function]}
      >
        <div
          className="react-date-picker__wrapper"
        >
          <div
            className="react-date-picker__inputGroup"
            onClick={[Function]}
          >
            <input
              max="2021-10-06"
              min="1995-06-16"
              name="date"
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "position": "absolute",
                  "visibility": "hidden",
                  "zIndex": "-999",
                }
              }
              type="date"
              value="2021-10-06"
            />
            
            <input
              autoComplete="off"
              className="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
              data-input="true"
              inputMode="numeric"
              max={10}
              min={1}
              name="month"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder="--"
              required={null}
              type="number"
              value="10"
            />
            <span
              className="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              aria-label="Day"
              autoComplete="off"
              autoFocus={false}
              className="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
              data-input="true"
              inputMode="numeric"
              max={6}
              min={1}
              name="day"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder="--"
              required={null}
              type="number"
              value="6"
            />
            <span
              className="react-date-picker__inputGroup__divider"
            >
              /
            </span>
            <input
              autoComplete="off"
              autoFocus={false}
              className="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
              data-input="true"
              inputMode="numeric"
              max={2021}
              min={1995}
              name="year"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder="----"
              required={null}
              step={1}
              type="number"
              value="2021"
            />
            
          </div>
          <button
            className="react-date-picker__clear-button react-date-picker__button"
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <svg
              className="react-date-picker__clear-button__icon react-date-picker__button__icon"
              height={19}
              stroke="black"
              strokeWidth={2}
              viewBox="0 0 19 19"
              width={19}
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="4"
                x2="15"
                y1="4"
                y2="15"
              />
              <line
                x1="15"
                x2="4"
                y1="4"
                y2="15"
              />
            </svg>
          </button>
          <button
            className="react-date-picker__calendar-button react-date-picker__button"
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <svg
              className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
              height={19}
              stroke="black"
              strokeWidth={2}
              viewBox="0 0 19 19"
              width={19}
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="15"
                width="15"
                x="2"
                y="2"
              />
              <line
                x1="6"
                x2="6"
                y1="0"
                y2="4"
              />
              <line
                x1="13"
                x2="13"
                y1="0"
                y2="4"
              />
            </svg>
          </button>
        </div>
      </div>
    </section>
  </main>,
  <section
    className="sc-hKFxyN jLOHUB"
  >
    <img
      alt="Cartoon character of Astronaut in Space"
      src="http://images.all-free-download.com/images/graphiclarge/space_background_astronaut_spaceship_flag_icons_ornament_6828811.jpg"
    />
  </section>,
]
`;
